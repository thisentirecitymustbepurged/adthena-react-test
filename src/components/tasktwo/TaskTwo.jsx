import React from 'react';

const TaskTwo = () => (
  <div className="task">
    <h1>Task Two</h1>
    <div className="content">
      <h4>Complete the following task:</h4>
      <p>
          The task is to create components to fetch public API data, combine it, apply
          filtering and visualise that data.
      </p>
      <p>Include unit tests. Jest is already configured for you in the skeleton project.</p>
      Expected:
      <ol type="1">
        <li>
          Use data from the following API endpoints:
          <ol type="a">
            <li>https://jsonplaceholder.typicode.com/users</li>
            <li>https://jsonplaceholder.typicode.com/todos</li>
          </ol>
        </li>
        <li>
          Data fetching should be done with a re-usable data fetching hook.
        </li>
        <li>
          The user interface must accept a username as input.
        </li>
        <li>
          The user interface must handle the case where an error occurs and the case where the
          requested user is not found.
        </li>
        <li>
          Display the username, email and website of the user.
        </li>
        <li>
          Display a list of Todos for user.
        </li>
        <li>
          Visualise the Todos in such a way that it is easy to distinguish between the
          complete and incomplete Todos.
        </li>
        <li>
          Give the user the option of a dark theme for the Todos user interface.
        </li>
      </ol>
      <strong>Feel free to use this component for your implementation.</strong>
    </div>
  </div>
);

export default TaskTwo;
