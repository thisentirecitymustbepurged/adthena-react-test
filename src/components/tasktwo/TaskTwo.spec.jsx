import React from 'react';
import { shallow } from 'enzyme';
import TaskTwo from '@components/tasktwo/TaskTwo';

const render = props => shallow(<TaskTwo {...props} />);

describe('Task two component', () => {
  it('should match snapshot', () => {
    const wrapper = render({});
    expect(wrapper).toMatchSnapshot();
  });
});
