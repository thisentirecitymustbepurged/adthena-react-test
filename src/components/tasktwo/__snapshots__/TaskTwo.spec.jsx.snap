// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task two component should match snapshot 1`] = `
<div
  className="task"
>
  <h1>
    Task Two: Todos
  </h1>
  <div
    className="content"
  >
    <Loader
      itemsToLoad={
        Object {
          "todos": [Function],
          "users": [Function],
        }
      }
    >
      <Users
        todos={Array []}
        users={Array []}
      />
    </Loader>
  </div>
</div>
`;
